code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -392,23 +392,52 @@
 
 /obj/item/tool/weldpack/attackby(obj/item/I, mob/user, params)
 	. = ..()
+	if(reagents.total_volume == 0)
+		to_chat(user, "<span class='warning'>Out of fuel!</span>")
+		return
 
-	if(iswelder(I))
+	else if(iswelder(I))
```

Comment:
Could you move this to `welder_act()`?

Reply:
Any reason why? Other backpacks have refueling under them.

Reply:
@XElectricX `welder_act` just allows you to move this to a specific proc call and share code.
It would be a good change.